docs/src/main/sphinx/optimizer/cost-based-optimizations.rst

```diff
@@ -10,54 +10,53 @@ Join enumeration
+automatically picks the join order with the lowest computed costs.
 
 The join enumeration strategy is governed by the ``join_reordering_strategy``
 session property, with the ``optimizer.join-reordering-strategy``
 configuration property providing the default value.
 
-The valid values are:
- * ``AUTOMATIC`` (default) - full automatic join enumeration enabled
+The possible values are:
+ * ``AUTOMATIC`` (default) - enable full automatic join enumeration
  * ``ELIMINATE_CROSS_JOINS`` - eliminate unnecessary cross joins
  * ``NONE`` - purely syntactic join order
 
-If using ``AUTOMATIC`` and statistics are not available, or if for any other
-reason a cost could not be computed, the ``ELIMINATE_CROSS_JOINS`` strategy is
-used instead.
+If you are using ``AUTOMATIC`` join enumeration and statistics are not
+available or a cost can not be computed for any other reason, the
+``ELIMINATE_CROSS_JOINS`` strategy is used instead.
 
 Join distribution selection
 ---------------------------
 
-Trino uses a hash based join algorithm. That implies that for each join
-operator a hash table must be created from one join input, called build side.
-The other input, called probe side, is then iterated, and for each row the hash table is
-queried to find matching rows.
+Trino uses a hash-based join algorithm. For each join operator, a hash table
+must be created from one join input, referred to as the build side. The other
+input, called the probe side, is then iterated on. For each row, the hash table
+is queried to find matching rows.
 
 There are two types of join distributions:
- * Partitioned: each node participating in the query builds a hash table
-   from only fraction of the data
- * Broadcast: each node participating in the query builds a hash table
-   from all of the data (data is replicated to each node)
-
-Each type have their trade offs. Partitioned joins require redistributing both
-tables using a hash of the join key. This can be slower, sometimes
-substantially slower, than broadcast joins, but allows much larger joins. In
-particular, broadcast joins are faster if the build side is much smaller
-than the probe side. However, broadcast joins require that the tables on the
-build side of the join after filtering fit in memory on each node, whereas
-distributed joins only need to fit in distributed memory across all nodes.
-
-With cost based join distribution selection, Trino automatically chooses to
-use a partitioned or broadcast join. With cost based join enumeration, Trino
-automatically chooses which side is the probe and which is the build.
+ * Partitioned: each node participating in the query builds a hash table from
+   only fraction of the data
+ * Broadcast: each node participating in the query builds a hash table from all
+   of the data (data is replicated to each node)
+
+Each type has its trade offs. Partitioned joins require redistributing both
+tables using a hash of the join key. This can be slower or much slower than
```

Comment:
"this can be slower or much slower" seems redundant, why not just say "can be much slower"?